### PR TITLE
Update documentation - restructure the documentation and include more examples

### PR DESCRIPTION
**Pull request by vedran-kasalica**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+jobs:
+  build-docs:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Check out the repo
+      uses: actions/checkout@v2
+
+    - name: Set up Python
+      uses: actions/setup-python@v2
```
Comment (kretep) [inline .github/workflows/build.yml:17]: don't think it matters much, but I think these actions have more recent versions

File: docs/demo/geo_gmt/geo_gmt.rst [hunk 2]
```diff
@@ -71,7 +76,7 @@ the domain [1]_.
 .. code-block:: shell
 
     cd ~/git/APE_UseCases
-    java -jar APE-<version>.jar GeoGMT/E1/ape.configuration
+    java -jar APE-<version>.jar GeoGMT/E1/config.jsonuration
```
Comment (kretep) [inline docs/demo/geo_gmt/geo_gmt.rst:79]: This might be a typo

**Review by kretep — APPROVED** [review]
Looks fine overall; see comments for a few minor points.

File: docs/demo/geo_gmt/geo_gmt.rst (hunk 2 above)
Comment (vedran-kasalica) [reply docs/demo/geo_gmt/geo_gmt.rst]: It was, thanks!

File: .github/workflows/build.yml (hunk 1 above)
Comment (vedran-kasalica) [reply .github/workflows/build.yml]: I updated them, thanks

**Pull request merged**